tests/cnf/core/network/metallb/tests/metallb-crds.go

```diff
@@ -0,0 +1,190 @@
+package tests
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/metallb"
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
+
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/ipaddr"
+	. "github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/netinittools"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/netparam"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/cmd"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/metallbenv"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/tsparams"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var _ = Describe("MetalLb New CRDs", Ordered, Label("newcrds"), ContinueOnFailure, func() {
+	var (
+		ipv4metalLBIPList        []string
+		addressPool              = []string{"3.3.3.1", "3.3.3.240"}
+		ipAddressPool            *metallb.IPAddressPoolBuilder
+		l3ClientPod              *pod.Builder
+		sriovInterfacesUnderTest []string
+		ipSecondaryInterface1    = "3.3.3.10/24"
+		ipSecondaryInterface2    = "3.3.3.20/24"
+		cnfWorkerNodeList        []*nodes.Builder
+	)
+
+	BeforeAll(func() {
+		By("Collecting information before test")
+		masterNodeList, err := nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.ControlPlaneLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "An unexpected error occurred while getting master nodes.")
+		Expect(len(masterNodeList)).To(BeNumerically(">", 0),
+			"Master node list is empty")
+		firstMasterNode := masterNodeList[0]
+
+		cnfWorkerNodeList, err = nodes.List(APIClient,
```

Comment:
Maybe add some more By() stating what is happening like
		By("Getting external nodes ip addresses")
		By("Selecting worker node for BGP tests")

Reply:
added